src/main/java/org/icij/datashare/Neo4jResource.java

```diff
@@ -378,12 +379,18 @@ private <T> Payload wrapNeo4jAppCall(Provider<T> importProvider) {
         }
     }
 
-    private void checkAccess(String project, Context context) {
+    protected void checkAccess(String project, Context context) {
         if (!((User) context.currentUser()).isGranted(project)) {
             throw new UnauthorizedException();
         }
     }
 
+    protected void checkCheckLocal(Context context) {
+        if (!((User) context.currentUser()).isLocal()) {
```

Comment:
this test whether the user is called "local".

https://github.com/ICIJ/datashare-api/blob/c1efd9381581f97ccd3c2f29103fb356cd8ae7f7/src/main/java/org/icij/datashare/user/User.java#L108

So if in `SERVER`mode a user is called "local" it will pass.

I'm not sure if it the intent. But to check that it is in `LOCAL` mode it would be better to test the mode (with PropertiesProvider).

Reply:
Thank you. I could is checking the `String` enough or should I move the `Mode` enum to the `datashare-api` and check the string property against the `Mode` enum ?

Reply:
Get the `mode` property from `PropertiesProvider`

Reply:
Yes you can check the propertiesProvider by looking the `mode` key, and if it's "LOCAL" or "EMBEDDED" we would allow it